pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -470,6 +491,26 @@ func (r *ReconcileJenkinsBaseConfiguration) createService(meta metav1.ObjectMeta
 	return stackerr.WithStack(r.UpdateResource(&service))
 }
 
+func (r *ReconcileJenkinsBaseConfiguration) createRoute(meta metav1.ObjectMeta, name string) error {
+	route := routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: meta.Namespace,
+			Labels:    meta.Labels,
+		},
+	}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: meta.Namespace}, &route)
+	if err != nil && apierrors.IsNotFound(err) {
+		resources.UpdateRouteFromService(&route, name)
```

Comment:
shouldn't we simply remove the route and recreate it if the Service changed ?

Reply:
UpdatRouteFromService changed to PopulateRouteFromService, cuz we are not really updating here